pkg/router/route.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package router
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+import (
+	"github.com/emirpasic/gods/trees/avltree"
+	"github.com/pkg/errors"
+)
+
+import (
+	"github.com/dubbogo/dubbo-go-proxy/pkg/common/constant"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/config"
+)
+
+// Node defines the single method of the router configured API
+type Node struct {
+	fullPath string
+	wildcard bool
+	methods  map[config.HTTPVerb]*config.Method
+}
+
+// Route defines the tree of router APIs
+type Route struct {
+	lock         sync.RWMutex
```

Comment:
Remember: if this object has a lock, dont copy this object, it will cause a lock copy problem.

Ref: https://github.com/apache/dubbo-go/issues/511